CMakeLists.txt

```diff
@@ -95,7 +93,7 @@ target_include_directories(pisa INTERFACE external)
 
 add_subdirectory(src)
 
-if (BUILD_TESTING AND NOT_SUBPROJECT)
+if (PISA_ENABLE_TESTING)
```

Comment:
`BUILD_TESTING` is a standard way of turning off test for CTest: https://cmake.org/cmake/help/v3.13/module/CTest.html

Reply:
Yes I am aware, but why do we need it?
`BUILD_TESTING` is always `ON`. If you include pisa as a subproject you will eventually set `PISA_ENABLE_TESTING` to off. How can `BUILD_TESTING` help?

Reply:
`BUILD_TESTING` is supposed to override any other options. So I would change it to:
```
if (PISA_ENABLE_TESTING AND BUILD_TESTING)
```